app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharecterView.kt

```diff
@@ -1,53 +1,71 @@
 package com.puzzlebench.clean_marvel_kotlin.presentation.mvp
 
-import android.support.v7.widget.GridLayoutManager
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
 import android.view.View
 import com.puzzlebench.clean_marvel_kotlin.R
 import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
-import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_main.recycleView
+import kotlinx.android.synthetic.main.activity_main.progressBar
 import java.lang.ref.WeakReference
 
-class CharecterView(activity: MainActivity) {
+
+class CharacterView(activity: MainActivity) {
+
+    companion object {
+        const val EXTRA_CHARACTER = "character"
+        private const val TAG = "txn_tag"
+    }
+
     private val activityRef = WeakReference(activity)
-    private val SPAN_COUNT = 1
-    var adapter = CharacterAdapter { character -> activity.applicationContext.showToast(character.name) }
+    var adapter = CharacterAdapter(emptyList()) { character ->
+                activity.applicationContext.showToast(character.name)
+                activity.getCharacterById(character.id)
+    }
 
     fun init() {
         val activity = activityRef.get()
-        if (activity != null) {
-            activity.recycleView.layoutManager = GridLayoutManager(activity, SPAN_COUNT)
-            activity.recycleView.adapter = adapter
+        activity?.let { act ->
+            act.recycleView.layoutManager = LinearLayoutManager(act.applicationContext, RecyclerView.VERTICAL, false)
+            act.recycleView.adapter = adapter
             showLoading()
         }
-
     }
 
     fun showToastNoItemToShow() {
         val activity = activityRef.get()
-        if (activity != null) {
-            val message = activity.baseContext.resources.getString(R.string.message_no_items_to_show)
-            activity.applicationContext.showToast(message)
-
+        activity?.let { act ->
```

Comment:
if you are referring to the activity you can use "it"

Reply:
Done!